pipeline_steps/common.groovy

```diff
@@ -439,24 +449,72 @@ def prepareRpcGit(String branch = "auto", String dest = "/opt"){
 
     print("Repo: ${env.RPC_REPO} Branch: ${branch}")
 
-    // checkout used instead of git as a custom refspec is required
-    // to checkout pull requests
-    checkout([$class: 'GitSCM',
-      branches: [[name: branch]],
-      doGenerateSubmoduleConfigurations: false,
-      extensions: [[$class: 'CleanCheckout']],
-      submoduleCfg: [],
-      userRemoteConfigs: [
-        [
-          url: env.RPC_REPO,
-          refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
-        ]
-      ]
-    ]) // checkout
-    sh "git submodule update --init"
+    clone_with_pr_refs(env.RPC_REPO, branch)
   } // dir
 }
 
+// Clone repo with Refspecs required for PRs.
+// Should't need to supply any params to checkout a PR merged with the base.
+// Uses shell+git to avoid hostname verification failures with
```

Comment:
I got the credentials working with the checkout step, the problem was host key verification, not user key/creds. The solutions are mostly "add this hostkey to the jenkins user's known hosts" but I don't want jobs making changes outside of their workspace and I don't want to have to do any manual setup of slaves. Using shell+git to clone keeps all the config within the workspace. 

I guess we could have something  in the long-running-slave job, or in the setup-jenkins-slave playbook that added the github host keys to the jenkins user ? But i don't see whats so wrong with shelling out to git, thats what checkout + git plugin does anyway 🤷‍♂️